src/Core/src/Platform/Windows/SearchBarExtensions.cs

```diff
@@ -16,110 +35,85 @@ public static void UpdatePlaceholder(this AutoSuggestBox platformControl, ISearc
 			platformControl.PlaceholderText = searchBar.Placeholder ?? string.Empty;
 		}
 
-		public static void UpdatePlaceholderColor(this AutoSuggestBox platformControl, ISearchBar searchBar, Brush? defaultPlaceholderColorBrush, Brush? defaultPlaceholderColorFocusBrush, MauiSearchTextBox? queryTextBox)
+		private static readonly string[] _placeholderForegroundColorKeys =
 		{
-			if (queryTextBox == null)
-				queryTextBox = platformControl.GetFirstDescendant<MauiSearchTextBox>();
-
-			if (queryTextBox == null)
-				return;
+			"TextControlPlaceholderForeground",
+			"TextControlPlaceholderForegroundPointerOver",
+			"TextControlPlaceholderForegroundFocused",
+			"TextControlPlaceholderForegroundDisabled"
+		};
 
-			Color placeholderColor = searchBar.PlaceholderColor;
-			if (placeholderColor != null)
-				queryTextBox.PlaceholderForegroundBrush = queryTextBox.PlaceholderForegroundFocusBrush = placeholderColor.ToPlatform();
+		public static void UpdatePlaceholderColor(this AutoSuggestBox platformControl, ISearchBar searchBar)
+		{
+			UpdateColors(platformControl.Resources, _placeholderForegroundColorKeys,
+				searchBar.PlaceholderColor?.ToPlatform());
 		}
 
 		public static void UpdateText(this AutoSuggestBox platformControl, ISearchBar searchBar)
 		{
 			platformControl.Text = searchBar.Text;
 		}
 
-		public static void UpdateTextColor(this AutoSuggestBox platformControl, ISearchBar searchBar, Brush? defaultTextColorBrush, Brush? defaultTextColorFocusBrush, MauiSearchTextBox? queryTextBox)
+		private static readonly string[] _foregroundColorKeys =
 		{
-			if (queryTextBox == null)
-				queryTextBox = platformControl.GetFirstDescendant<MauiSearchTextBox>();
+			"TextControlForeground",
+			"TextControlForegroundPointerOver",
+			"TextControlForegroundFocused",
+			"TextControlForegroundDisabled"
+		};
 
-			if (queryTextBox == null)
-				return;
+		public static void UpdateTextColor(this AutoSuggestBox platformControl, ISearchBar searchBar)
+		{
+			UpdateColors(platformControl.Resources, _foregroundColorKeys, searchBar.TextColor?.ToPlatform());
+		}
 
-			Color textColor = searchBar.TextColor;
-			if (textColor != null)
-				queryTextBox.Foreground = queryTextBox.ForegroundFocusBrush = textColor.ToPlatform();
+		private static void UpdateColors(ResourceDictionary resource, string[] keys, Brush? brush)
+		{
+			if (brush is null)
+			{
+				foreach (var key in keys)
+				{
+					resource.Remove(key);
+				}
+			}
+			else
+			{
+				foreach (var key in keys)
+				{
+					resource[key] = brush;
+				}
```

Comment:
Nit: #5479 added utility methods `ResourceDictionaryExtensions.RemoveKeys/SetValueForAllKey` for these.